### DISCUSSION TITLE
No Wifi on Macbook Pro mid 2014

### DISCUSSION THREAD

**Discussion by michelestaffiere** [elementary/triage#252] [Q&A]
tried everything from removing and reinstalling broadcom drivers from the App center and through the terminal but nothing works.

Installed on MacBook pro mid 2014

`03:00.0 Network Controller Broadcom Inc. and subsidiaries BCM4360 802.11ac Wireless Network Adapter (Rev 03)`

please help, have tried numerous "fixes" but nothing works.

**Comment by lenemter** [comment]
In Google people recommend installing `bcmwl-kernel-source` package. Have you tried it?

Paste in terminal:
```
sudo apt install bcmwl-kernel-source
```

**Reply by stam66** [reply]
I had the same issue testing on a 2016 MBP - I could only get wifi if I put the laptop right next to the router which is weird...
I ended up abandoning.

Did the bcmwl install help revive wifi?

**Comment by RogerWeihrauch** [comment]
Hi to all
Maybe it helps to look for a kind of kernel module/its source on a 'broadcom-sta' instead a broadcom/bcmwl kernel sources?

Since Elementary OS is based on Ubuntu and thereas also on Debian, I succeeded to get my Wifi Working on a MacBookPro Mid20212 in Debian 10/11/12.

Regards,
Roger
